sql/memo/rel_props.go

```diff
@@ -52,13 +53,257 @@ func newRelProps(rel RelExpr) *relProps {
 		} else {
 			p.outputCols = r.OutputCols()
 		}
+		// need to assign column ids
+		// TODO name resolution should replace assignColumnIds, then Fds can stay lazy
+		p.populateFds()
 	}
 	p.populateOutputTables()
 	p.populateInputTables()
-
 	return p
 }
 
+// denormIdxExprs replaces the native table name in index
+// expression strings with the aliased name.
+// TODO: this is unstable as long as periods in Index.Expressions()
+// identifiers are ambiguous.
+func denormIdxExprs(table string, idx sql.Index) []string {
+	denormExpr := make([]string, len(idx.Expressions()))
+	for i, e := range idx.Expressions() {
+		parts := strings.Split(e, ".")
+		denormExpr[i] = strings.ToLower(fmt.Sprintf("%s.%s", table, parts[1]))
+	}
+	return denormExpr
+}
+
+func (p *relProps) populateFds() {
```

Comment:
Same comment about unit tests for this function

Doesn't need to be extensive, but having ~1 unit test for each case in the switch statement is going to make this a lot easier to understand for the next guy